api/src/schedules/metrics.ts

```diff
@@ -0,0 +1,46 @@
+import { useEnv } from '@directus/env';
+import { toBoolean } from '@directus/utils';
+import { scheduleJob } from 'node-schedule';
+import { useMetrics } from '../metrics/index.js';
+import { validateCron } from '../utils/schedule.js';
+
+const METRICS_LOCK_TIMEOUT = 10 * 60 * 1000; // 10 mins
+
+const metrics = useMetrics();
+let lockedAt = 0;
+
+export async function handleMetricsJob() {
+	const now = Date.now();
+
+	if (lockedAt !== 0 && lockedAt > now - METRICS_LOCK_TIMEOUT) {
+		// ensure only generating metrics once per node
+		return;
+	}
+
+	lockedAt = Date.now();
+
+	await metrics.generate();
+
+	lockedAt = 0;
```

Comment:
We should put this in a `try...finally` block to ensure lockedAt is reset even if the generation fails for some reason:

```suggestion
	try {
		await metrics.generate();
	} finally {
		lockedAt = 0;
	}
```

Reply:
Agreed!

Reply:
Per discussion update to use catch in addition to finally via ee0d44d0b7fc68976351b6661186bfb035c4e330